enterprise/api/v1/views.py

```diff
@@ -82,6 +84,26 @@ class EnterpriseCustomerUserViewSet(EnterpriseReadOnlyModelViewSet):
     filter_fields = FIELDS
     ordering_fields = FIELDS
 
+    @detail_route()
+    def entitlements(self, request, pk=None):  # pylint: disable=invalid-name,unused-argument
+        """
+        Retrieve the list of entitlements available to this learner.
+
+        Only those entitlements are returned that satisfy enterprise customer's data sharing setting.
+
+        Arguments:
+            request (HttpRequest): Reference to in progress request instance.
+            pk (Int): value if the primary key for the selected enterprise learner
+
+        Returns:
+            (HttpResponse): Response object containing a list of learner's entitlements.
+        """
+        enterprise_customer_user = self.get_object()
```

Comment:
details view creates a url something like `/enterprise/api/v1/enterprise-learner/{pk}/entitlements/`, so, if a record does not exist drf will return 404 before even executing this code.